packages/hopli/src/key_pair.rs

```diff
@@ -62,20 +69,21 @@ pub fn create_identity(dir_name: &str, password: &str, maybe_name: &Option<Strin
         None => format!("{dir_name}/{}.id", { keys.id().to_string() }),
     };
 
-    if Path::new(&file_path).exists() {
+    let path = Path::new(&file_path);
+    if path.exists() {
         return Err(HelperErrors::IdentityFileExists(file_path));
     } else {
         keys.write_eth_keystore(&file_path, password, false)?;
     }
 
-    Ok(keys)
+    Ok((String::from(path.file_name().unwrap().to_str().unwrap()), keys))
```

Comment:
Why `unwrap` and not `?` ?

Reply:
Because I did not know better.  Will change it.